examples/pytorch/graphsaint/sampler.py

```diff
@@ -10,23 +10,74 @@
 
 
 # The base class of sampler
-# (TODO): online sampling
-class SAINTSampler(object):
-    def __init__(self, dn, g, train_nid, node_budget, num_repeat=50):
-        """
-        :param dn: name of dataset
-        :param g: full graph
-        :param train_nid: ids of training nodes
-        :param node_budget: expected number of sampled nodes
-        :param num_repeat: number of times of repeating sampling one node
-        """
-        self.g = g
+class SAINTSampler:
+    """
+    Description
+    -----------
+    SAINTSampler implements the sampler described in GraphSAINT. This sampler implements offline sampling in
+    pre-sampling phase as well as fully offline sampling, fully online sampling in training phase.
+    Users can conveniently set param 'online' of the sampler to choose different modes.
+
+    Parameters
+    ----------
+    dn : str
+        name of dataset.
+    g : DGLGraph
+        the full graph.
+    train_nid : list
+        ids of training nodes.
+    num_workers_sampler : int
+        number of processes to sample subgraphs in pre-sampling procedure using torch.dataloader.
+    train : bool
+        a flag specifying if the sampler is utilized in training procedure or not. Note that the sampling methods are
+        different in pre-sampling phase and training phase respectively. We have to assign this flag as False before
+        pre-sampling phase and True after pre-sampling before training.
```

Comment:
We don't need a user to set the flag since we always perform pre-sampling in class initialization.